app/org/sagebionetworks/bridge/upload/GenericUploadFormatHandler.java

```diff
@@ -0,0 +1,217 @@
+    UploadSchema getUploadSchema(StudyIdentifier studyId, JsonNode infoJson) throws UploadValidationException {
+        // Try getting by survey first.
+        String surveyGuid = JsonUtils.asText(infoJson, UploadUtil.FIELD_SURVEY_GUID);
+        String surveyCreatedOn = JsonUtils.asText(infoJson, UploadUtil.FIELD_SURVEY_CREATED_ON);
+        if (StringUtils.isNotBlank(surveyGuid) && StringUtils.isNotBlank(surveyCreatedOn)) {
+            // surveyCreatedOn is a timestamp. SurveyService takes long epoch millis. Convert.
+            long surveyCreatedOnMillis= DateUtils.convertToMillisFromEpoch(surveyCreatedOn);
+
+            // Get survey. We use the survey identifier as the schema ID and the schema revision. Both of these must be
+            // specified.
+            Survey survey = surveyService.getSurvey(new GuidCreatedOnVersionHolderImpl(surveyGuid,
+                    surveyCreatedOnMillis));
+            String surveySchemaId = survey.getIdentifier();
+            Integer surveySchemaRev = survey.getSchemaRevision();
+            if (StringUtils.isBlank(surveySchemaId) || surveySchemaRev == null) {
+                throw new UploadValidationException("Schema not found for survey " + surveyGuid + ":" +
+                        surveyCreatedOnMillis);
+            }
+
+            // Get the schema with the schema ID and rev.
+            return uploadSchemaService.getUploadSchemaByIdAndRev(studyId, surveySchemaId, surveySchemaRev);
+        }
+
+        // Fall back to getting by schema.
+        String schemaId = JsonUtils.asText(infoJson, UploadUtil.FIELD_ITEM);
+        Integer schemaRev = JsonUtils.asInt(infoJson, UploadUtil.FIELD_SCHEMA_REV);
+        if (StringUtils.isNotBlank(schemaId) && schemaRev != null) {
+            return uploadSchemaService.getUploadSchemaByIdAndRev(studyId, schemaId, schemaRev);
+        } else {
+            throw new UploadValidationException("info.json must contain either item and schemaRevision or " +
+                    "surveyGuid and surveyCreatedOn");
+        }
+    }
+
+    // Helper method to read and parse the createdOn from info.json and add it to the health data record. Handles
+    // fall-back logic.
+    // Package-scoped to facilitate unit tests.
+    static void parseCreatedOnToRecord(UploadValidationContext context, JsonNode infoJson, HealthDataRecord record) {
+        // createdOn string from info.json
+        String createdOnString = JsonUtils.asText(infoJson, UploadUtil.FIELD_CREATED_ON);
+
+        // Parse into a Joda DateTime.
+        DateTime createdOn = null;
+        if (StringUtils.isNotBlank(createdOnString)) {
+            try {
+                createdOn = DateTime.parse(createdOnString);
+            } catch (IllegalArgumentException ex) {
+                // Write a message to the validation context, but there's no need to log or throw.
+                context.addMessage("Invalid date-time: " + createdOnString);
+            }
+        }
```

Comment:
Could use BridgeUtils.getDateTimeOrDefault(createdOnString, null), and when null, write context message. Handles blank value and try/catch

Reply:
BridgeUtils.getDateTimeOrDefault() throws on parse error, so it'd be the same amount of code either way.